cql/fhir-cql/src/main/java/com/ibm/fhir/cql/engine/model/FHIRModelResolver.java

```diff
@@ -102,7 +103,13 @@ public String getPackageName() {
 
         // add all code subtypes
         for (Class<?> codeSubtype : ModelSupport.getCodeSubtypes()) {
+            if (codeSubtype == ResourceTypeCode.class) {
+                // special case handling for ResourceTypeCode
+                typeMap.put("ResourceType", codeSubtype);
+                typeMap.put("FHIRResourceType", codeSubtype);
+            }
```

Comment:
> Is that your thinking here?

precisely

> Are there other places where clients of the model would expect those old type names?

I hope not.  but thats a big reason I wanted to do this change in our 5.0 release...I feel a bit better about Java api-breaking changes like this to fhir-model if they are in a major release bump

> My first thought when I encountered this was that the mapping from our internal representation to the external FHIR representation is logic that might be more properly placed in ModelSupport.

Not exactly sure what you mean by that, but happy to discuss.

Reply:
Basically, I was wondering if ModelSupport should have something like `isResourceTypeCode(Class<?>)` or something like that. A utility that could be reused if other code outside of the FHIRModelResolver cared about this change. If the CQL engine is the only player, then it doesn't matter. I am ok with these changes (I don't have formal approval rights).

Reply:
OK.  This is the only spot in our codebase that would benefit from such a method.  And I'm hoping most folks aren't dependent on the class names of our code subtypes because they aren't really even real artifacts from FHIR (we derive them from the binding names on ElementDefinitions shipped with the spec). I think its mostly just luck/coincidence that we happened to use the same field name as HAPI for these Classnames in the first place.